smart-py/marketplace.py

```diff
@@ -54,12 +59,70 @@ def cancel_swap(self, params):
     def update_fee(self, params):
         sp.verify(sp.sender == self.data.manager)
         self.data.fee = params
-        
+
     @sp.entry_point
-    def update_manager(self, params):
+    def update_fee_account(self, params):
         sp.verify(sp.sender == self.data.manager)
+        self.data.fee_account= params
+
+    @sp.entry_point
+    def update_manager(self, params):
+        sp.if sp.now - self.data.last_ping > 100 * 24 * 3600:
+            sp.verify(self.data.secondary_accounts.contains(sp.sender))
+        sp.else:
+            sp.verify(sp.sender == self.data.manager)
+
         self.data.manager = params
-        
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def add_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def remove_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.remove(params)
+
+    @sp.entry_point
+    def ping(self):
```

Comment:
so the manager just calls this manually?

Reply:
Yes. He can call it every week or so. One could also update the last_ping parameter in other entrypoints executed by the manager